docs/userstories/v1/egress-control-in-anp/egress-control-cluster-nodes.md

```diff
@@ -0,0 +1,41 @@
+
+### Summary
+
+1. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to k8s-apiserver.
+2. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to specific ports on cluster nodes.
+
+### Description
+
+Currently the behaviour for policies defined around traffic from cluster
+workloads (non-hostNetworked pods) towards hostNetworked pods or nodes in the
+cluster is undefined. See https://github.com/kubernetes-sigs/network-policy-api/issues/73.
+
+ANP currently supports only east-west traffic and this traffic flow cuts from
+overlay to underlay which makes this part of the egress (south-north) use case.
+
+Let's provide a defined behaviour in ANP to explicitly achieve the use cases.
+
+### Acceptance Criteria
+
+1. If I apply a policy that limits egress access from workloads identified
+by label:A towards k8s-apiserver backends (master nodes at 6443 port), it should
+be able to guarantee that those workloads cannot access the apiserver
+endpoints:targetPort. Even if the master node IPs change, the policy should be
+able to automatically pivot towards the new nodeIPs.
+
+[
+Also, here are a few points that need to be addressed:
+
+1. Since ANP is a new API, define if our current namespace and pod selectors
+consider host-networked pods or not? Or do we want to allow this discretion to the
+CNI plugins?
```

Comment:
> IMO we should avoid implementation-defined behavior as much as possible. A policy object should have a single well-defined meaning, which is the same regardless of network plugin.

+1

> 
> The host-networked pod case is tricky because: (a) there are good use cases for being able to select host-network pods, but (b) most existing NetworkPolicy implementations are not able to distinguish traffic from different host-network pods so we can't just require it.
true.
> 
> One possibility would be to add explicit `hostNetworkPodSelector` fields in addition to the regular `podSelector`, and then have clear semantics for rejecting the policy if the plugin can't implement it.

yeah that's what I had in mind, since we still want to give users the ability to select host-net pods but also want clarity on what the API fields stand for (whether it includes host-net or just regular pods), we could just go with what you said, our regular `PodSelector` will select only CNI pods, and we introduce a new `hostNetworkPodSelector` that can be used for specifying host-net pods. `NamespacedPeer` and `NamespacedPodPeer` and `NamespacedPodSubject` can all be modified to include this field.

Reply:
fixed this up, let me know what you think.
/cc @astoycos @Dyanngg